### ISSUE TITLE
Support exemplars in Prometheus exporter

### ISSUE THREAD

**Issue by damemi** [open-telemetry/opentelemetry-go#3163]
### Problem Statement

The Prometheus exporter in the new metrics SDK should be updated to support exemplars

### Additional Context

Original PR for prometheus exporter: https://github.com/open-telemetry/opentelemetry-go/pull/3135/files#r966288412

**Comment by songjiayang** [comment]
yeally need this feature, thanks.

**Comment by MrAlias** [comment]
Blocked by adding exemplars to the SDK in #559

**Comment by MrAlias** [comment]
Discussed in the SIG meeting today: this might better fit in the post-GA metric SDK project. It needs to be verified that this can be added in a backwards compatible manner. If so, work to add it should be postponed until the specification has stabilized this section.

**Comment by kernelpanic77** [comment]
does this issue still require work ?

**Comment by MrAlias** [comment]
@kernelpanic77 it looks like it does: https://github.com/open-telemetry/opentelemetry-go/blob/d9d9507227b30ad83500ec59e5714d0314ac83be/exporters/prometheus/exporter.go#L252

Should I assign it to you?

**Comment by MrAlias** [comment]
cc @dashpole

**Comment by kernelpanic77** [comment]
> @kernelpanic77 it looks like it does: https://github.com/open-telemetry/opentelemetry-go/blob/d9d9507227b30ad83500ec59e5714d0314ac83be/exporters/prometheus/exporter.go#L252
> 
> Should I assign it to you?

Sure I can have a look @MrAlias.
Although don't have much idea about the exporter. I'll get back to you with any queries I have.

**Comment by dashpole** [comment]
@kernelpanic77 let me know if you need any help with this, as it is fairly complex.  For reference, the spec for this is: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md#exemplars-1

> [Exemplars](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#exemplars) on OpenTelemetry Histograms and Monotonic Sums SHOULD be converted to OpenMetrics exemplars. Exemplars on other OpenTelemetry data points MUST be dropped. ...  For Prometheus pull endpoints, only a single exemplar is able to be added to each bucket, so the largest exemplar from each bucket MUST be used, if attaching exemplars. If no exemplars exist on a bucket, the highest exemplar from a lower bucket MUST be used, even though it is a duplicate of another bucket's exemplar. OpenMetrics Exemplars MUST use the trace_id and span_id keys for the trace and span IDs, respectively. Timestamps MUST be added as timestamps on the OpenMetrics exemplar, and filtered_attributes MUST be added as labels on the OpenMetrics exemplar unless they would exceed the OpenMetrics [limit on characters](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#exemplars).

**Comment by kernelpanic77** [comment]
Hi @dashpole, 
Thanks for the reference, i'm having a look at the spec for this. I am still trying to get familiar with exporters. Will discuss with you otel-go SIG this week.
